swarm/network/stream/delivery.go

```diff
@@ -230,13 +230,21 @@ func (d *Delivery) handleChunkDeliveryMsg(ctx context.Context, sp *Peer, req int
 	switch r := req.(type) {
 	case *ChunkDeliveryMsgRetrieval:
 		msg = (*ChunkDeliveryMsg)(r)
-		// do not sync if peer that is sending us a chunk is closer to the chunk then we are
 		peerPO := chunk.Proximity(sp.ID().Bytes(), msg.Addr)
 		po := chunk.Proximity(d.kad.BaseAddr(), msg.Addr)
-		if peerPO > po {
-			mode = chunk.ModePutRequest
-		} else {
+		depth := d.kad.NeighbourhoodDepth()
+		// is the chunk within our area of responsibility?
+		if po < depth {
```

Comment:
````golang
if po >= depth || peerPO < po {
         mode = chunk.ModePutSync
} else {
         mode = chunk.ModePutRequest
}
```

Reply:
Yes, this is more compact representation. I thought that the current one was easier to reason about, but this is good with a comment, as well.

Reply:
that is good but you got the comparison wrong, note
`po >= depth` -> sync

Reply:
Thanks Viktor.